src/lib.rs

```diff
@@ -0,0 +1,1167 @@
+            panic!("Cannot 'and' two Vobs of different length ({}  {})", self.len, other.len);
+        }
+        let mut chngd = false;
+        for (sb, ob) in self.vec.iter_mut().zip(other.vec.iter()) {
+            let ov = *sb;
+            let nv = ov & *ob;
+            if ov != nv {
+                chngd = true;
+            }
+            *sb = nv;
+        }
+        self.mask_last_block();
+        chngd
+    }
+
+    /// For each bit in this Vob, `or` it with the corresponding bit in `other`, returning `true`
+    /// if this led to any changes or `false` otherwise. The two Vobs must have the same number of
+    /// bits.
+    ///
+    /// # Panics
+    ///
+    /// If the two Vobs are of different length.
+    ///
+    /// # Examples
+    /// ```
+    /// #[macro_use] extern crate vob;
+    /// fn main() {
+    ///     let mut v1 = vob![true, false, false];
+    ///     let v2 = vob![false, true, false];
+    ///     v1.or(&v2);
+    ///     assert_eq!(v1, vob![true, true, false]);
+    /// }
+    /// ```
+    pub fn or(&mut self, other: &Vob<T>) -> bool {
+        if self.len != other.len {
+            panic!("Cannot 'or' two Vobs of different length ({}  {})", self.len, other.len);
+        }
+        let mut chngd = false;
+        for (sb, ob) in self.vec.iter_mut().zip(other.vec.iter()) {
+            let ov = *sb;
+            let nv = ov | *ob;
+            if ov != nv {
+                chngd = true;
+            }
+            *sb = nv;
+        }
+        self.mask_last_block();
+        chngd
+    }
+
+    /// For each bit in this Vob, `xor` it with the corresponding bit in `other`, returning `true`
+    /// if this led to any changes or `false` otherwise. The two Vobs must have the same number of
+    /// bits.
+    ///
+    /// # Panics
+    ///
+    /// If the two Vobs are of different length.
+    ///
+    /// # Examples
+    /// ```
+    /// #[macro_use] extern crate vob;
+    /// fn main() {
+    ///     let mut v1 = vob![true, false, true];
+    ///     let v2 = vob![false, true, true];
+    ///     v1.xor(&v2);
+    ///     assert_eq!(v1, vob![true, true, false]);
+    /// }
+    /// ```
+    pub fn xor(&mut self, other: &Vob<T>) -> bool {
+        if self.len != other.len {
+            panic!("Cannot 'xor' two Vobs of different length ({}  {})", self.len, other.len);
+        }
+        let mut chngd = false;
+        for (sb, ob) in self.vec.iter_mut().zip(other.vec.iter()) {
+            let ov = *sb;
+            let nv = ov ^ *ob;
+            if ov != nv {
+                chngd = true;
+            }
+            *sb = nv;
+        }
+        self.mask_last_block();
+        chngd
+    }
+
+    /// We guarantee that the last storage block has no bits set past the "last" bit: this function
+    /// clears any such bits.
+    fn mask_last_block(&mut self) {
+        let db = self.len % bits_per_block::<T>();
+        if db > 0 {
+            let m = (T::one() << db) - T::one();
```

Comment:
`m`, `b`, `v`?

`m` might be a mask?

I know you like short variable names, but a this is a little hard to read.

Reply:
What does `db` stand for actually?

"do bits"?

Reply:
m = mask, b = block, v = value. I think the first two are, from context, pretty clear. The last one is, I admit, meaningless.

I also just realised that the three lines spell out "MBV" which is an amusing short version of a famous band ;)

Reply:
"dead bits". Should be "unused bits" or "ub". Will clarify.

Reply:
Yeah, i'm not a huge fan of these one-letter variables, but if you must.

I think this function is good, have some free tests:
```
    #[test]
    fn test_mask_last_block1() {
        let mut v = vob![true, true];
        assert_eq!(v.vec, vec![3]);

        v.vec[0] = 0xaaaaaaaa;
        v.len = 7;
        v.mask_last_block();
        assert_eq!(v.vec, vec![42]);

        v.len = 30;
        v.vec[0] = 0xffffaaaa;
        v.mask_last_block();
        assert_eq!(v.vec, vec![1073719978]);
    }

    #[test]
    fn test_mask_last_block2() {
        let mut v = Vob::from_elem(128, true);
        let full_block = 0xffffffffffffffff;
        assert_eq!(v.vec, vec![full_block, full_block]);

        let one_zero = 0xaaaaaaaaaaaaaaaa;
        v.len = 68;
        v.vec[0] = one_zero;
        v.vec[1] = v.vec[0];
        v.mask_last_block();
        assert_eq!(v.vec, vec![one_zero, 0b1010]);
    }
```

Reply:
I think these tests assume `u64` backing?

Reply:
yes, you'd need to polish them up if you wanted to integrate them.

I just wanted to be sure that this function did what it claimed and used this code.

Reply:
The first works with 32 bit storage; the second doesn't. Fixing...

Reply:
Ah, it just so happens I didn't touch the top 32 bits in the first, yes. It should work on both.